test/Conversion/FIRRTLToHW/lower-to-hw.mlir

```diff
@@ -1248,4 +1248,153 @@ firrtl.circuit "Simple"   attributes {annotations = [{class =
     firrtl.connect %out, %reset : !firrtl.uint<1>, !firrtl.uint<1>
   }
+
+  // CHECK-LABEL: hw.module @init1DVector
+  firrtl.module @init1DVector(in %clock: !firrtl.clock, in %a: !firrtl.vector<uint<1>, 2>, out %b: !firrtl.vector<uint<1>, 2>) {
+    %r = firrtl.reg %clock  : !firrtl.vector<uint<1>, 2>
+    // CHECK:      sv.ifdef.procedural "RANDOMIZE_REG_INIT"  {
+    // CHECK-NEXT:   %1 = sv.array_index_inout %r[%false] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %2 = comb.extract %RANDOM from 0 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %1, %2 : i1
+    // CHECK-NEXT:   %3 = sv.array_index_inout %r[%true] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM_0 = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %4 = comb.extract %RANDOM_0 from 0 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %3, %4 : i1
+    // CHECK-NEXT: }
+
+    firrtl.connect %r, %a : !firrtl.vector<uint<1>, 2>, !firrtl.vector<uint<1>, 2>
+    firrtl.connect %b, %r : !firrtl.vector<uint<1>, 2>, !firrtl.vector<uint<1>, 2>
+    // CHECK:      sv.always posedge %clock  {
+    // CHECK-NEXT:   sv.passign %r, %a : !hw.array<2xi1>
+    // CHECK-NEXT: }
+    // CHECK-NEXT: hw.output %0 : !hw.array<2xi1>
+  }
+
+  // CHECK-LABEL: hw.module @init2DVector
+  firrtl.module @init2DVector(in %clock: !firrtl.clock, in %a: !firrtl.vector<vector<uint<1>, 2>, 2>, out %b: !firrtl.vector<vector<uint<1>, 2>, 2>) {
+    %r = firrtl.reg %clock  : !firrtl.vector<vector<uint<1>, 2>, 2>
+    // CHECK:      sv.ifdef.procedural "RANDOMIZE_REG_INIT"  {
+    // CHECK-NEXT:   %1 = sv.array_index_inout %r[%false] : !hw.inout<array<2xarray<2xi1>>>, i1
+    // CHECK-NEXT:   %2 = sv.array_index_inout %1[%false] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %3 = comb.extract %RANDOM from 0 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %2, %3 : i1
+    // CHECK-NEXT:   %4 = sv.array_index_inout %1[%true] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM_0 = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %5 = comb.extract %RANDOM_0 from 0 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %4, %5 : i1
+    // CHECK-NEXT:   %6 = sv.array_index_inout %r[%true] : !hw.inout<array<2xarray<2xi1>>>, i1
+    // CHECK-NEXT:   %7 = sv.array_index_inout %6[%false] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM_1 = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %8 = comb.extract %RANDOM_1 from 0 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %7, %8 : i1
+    // CHECK-NEXT:   %9 = sv.array_index_inout %6[%true] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM_2 = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %10 = comb.extract %RANDOM_2 from 0 : (i32) -> i1
```

Comment:
Don't these randoms get reused now?

Reply:
They will be reused. I was working on different branches.